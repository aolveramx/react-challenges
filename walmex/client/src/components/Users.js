import useFetch from '../hooks/useFetch'
import TableUsers from './TableUsers'

import { Table } from './styles/Home.styled'

const Users = () => {
  const url = `/users`
  const { isLoading, data: users, error } = useFetch(url)

  return (
    <div>
      {isLoading && <div>Loading...</div>}
      {error && <div>{console.log(error)}</div>}
      {users && (
        <Table>
          <thead>
            <tr>
              <th>User</th>
              <th>Age</th>
            </tr>
          </thead>
          <tbody>
            <TableUsers users={users} />
          </tbody>
        </Table>
      )}
    </div>
  )
}

export default Users
