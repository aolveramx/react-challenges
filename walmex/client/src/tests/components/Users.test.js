import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import Users from "../../components/Users"

describe('Tests in <Users />', () => {

 test('should render <Users />', () => {
   const wrapper = shallow(<Users />)
   expect(wrapper).toMatchSnapshot()
 });

})