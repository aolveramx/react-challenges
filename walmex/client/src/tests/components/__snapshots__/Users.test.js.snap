// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Users /> should render <Users /> 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
